bin/api-docs/gen-components-docs/markdown/index.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+const json2md = require( 'json2md' );
+
+/**
+ * Internal dependencies
+ */
+const { generateMarkdownPropsJson } = require( './props' );
+
+function generateMarkdownDocs( { typeDocs, subcomponentTypeDocs } ) {
+	const mainDocsJson = [
+		'<!-- This file is generated automatically and cannot be edited directly. -->\n',
+		{ h1: typeDocs.displayName },
+		{
+			p: `<p class="callout callout-info">See the <a href="https://wordpress.github.io/gutenberg/?path=/docs/components-${ typeDocs.displayName.toLowerCase() }--docs">WordPress Storybook</a> for more detailed, interactive documentation.</p>`,
+		},
+		typeDocs.description,
+		...generateMarkdownPropsJson( typeDocs.props ),
+	];
+
+	const subcomponentDocsJson = subcomponentTypeDocs
+		? [
+				{ h2: 'Subcomponents' },
+				...subcomponentTypeDocs.flatMap( ( subcomponentTypeDoc ) => [
+					{
+						h3: subcomponentTypeDoc.displayName,
```

Comment:
Are we convinced `subcomponentTypeDoc` will be well-formed and will contain all the fields we're using here?

Reply:
As far as I know, it should be sufficient for now.